driver-core/src/test/java/com/datastax/driver/core/UserTypesTest.java

```diff
@@ -416,139 +353,137 @@ public void udtNestedTest() throws Exception {
-        session().execute("USE testUdtsWithCollectionNulls");
-
+    public void should_save_and_retrieve_UDTs_with_null_collections() throws Exception {
         // create UDT
-        session().execute("CREATE TYPE user (a List<text>, b Set<text>, c Map<text, text>, d frozen<Tuple<text>>)");
-        session().execute("CREATE TABLE mytable (a int PRIMARY KEY, b frozen<user>)");
+        session().execute("CREATE TYPE user_null_collections (a List<text>, b Set<text>, c Map<text, text>, d frozen<Tuple<text>>)");
+        session().execute("CREATE TABLE null_collections_table (a int PRIMARY KEY, b frozen<user_null_collections>)");
 
         // insert null UDT data
-        PreparedStatement ins = session().prepare("INSERT INTO mytable (a, b) " +
+        PreparedStatement ins = session().prepare("INSERT INTO null_collections_table (a, b) " +
                 "VALUES (0, { a: ?, b: ?, c: ?, d: ? })");
         session().execute(ins.bind().setList(0, null).setSet(1, null).setMap(2, null).setTupleValue(3, null));
 
         // retrieve and verify data
-        ResultSet rs = session().execute("SELECT * FROM mytable");
+        ResultSet rs = session().execute("SELECT * FROM null_collections_table");
         List<Row> rows = rs.all();
-        assertEquals(1, rows.size());
+        assertThat(rows.size()).isEqualTo(1);
 
         Row row = rows.get(0);
-        assertEquals(row.getInt("a"), 0);
+        assertThat(row.getInt("a")).isEqualTo(0);
 
-        UserType userTypeDef = cluster().getMetadata().getKeyspace("testUdtsWithCollectionNulls").getUserType("user");
+        UserType userTypeDef = cluster().getMetadata().getKeyspace(keyspace).getUserType("user_null_collections");
         UDTValue userType = userTypeDef.newValue().setList("a", null).setSet("b", null).setMap("c", null).setTupleValue("d", null);
-        assertEquals(row.getUDTValue("b"), userType);
+        assertThat(row.getUDTValue("b")).isEqualTo(userType);
 
         // test missing UDT args
-        ins = session().prepare("INSERT INTO mytable (a, b) " +
+        ins = session().prepare("INSERT INTO null_collections_table (a, b) " +
                 "VALUES (1, { a: ? })");
         session().execute(ins.bind().setList(0, new ArrayList<Object>()));
 
         // retrieve and verify data
-        rs = session().execute("SELECT * FROM mytable");
+        rs = session().execute("SELECT * FROM null_collections_table");
         rows = rs.all();
-        assertEquals(2, rows.size());
+        assertThat(rows.size()).isEqualTo(2);
 
         row = rows.get(0);
-        assertEquals(row.getInt("a"), 1);
+        assertThat(row.getInt("a")).isEqualTo(1);
 
         userType = userTypeDef.newValue().setList(0, new ArrayList<Object>());
-        assertEquals(row.getUDTValue("b"), userType);
+        assertThat(row.getUDTValue("b")).isEqualTo(userType);
+    }
+
+    @Test(groups = "short")
+    @CassandraVersion("3.6")
+    public void should_indicate_if_user_type_is_frozen() {
```

Comment:
Minor nit, but what do you think about separating this into two separate tests, one that requires 2.1+ that creates a frozen udt and one that requires 3.6+ that creates a non-frozen UDT?  This will allow us to validate that the UDT is considered frozen for 2.1+.